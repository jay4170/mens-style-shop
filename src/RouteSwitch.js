import { BrowserRouter, Routes, Route } from "react-router-dom";
import HomePage from "./Home/HomePage";
import Shop from "./Shop/Shop";
import Header from "./Constants/Header";
import Footer from "./Constants/Footer";
import { useEffect, useState } from "react";
import fakeAPIcall from "./fakeApiCall";
import Basket from "./Basket/Basket";

const RouteSwitch = () => {
  const [products, setProducts] = useState([]);
  const [isOpen, setIsOpen] = useState(false);
  const [trolley, setTrolley] = useState([]);
  const [searchOpen, setSearchOpen] = useState(false);

  ///If the search icon is clicked set state to open and display search bar
  ///If anywhere on the page is clicked, hide open search bar
  const handleSearchOpening = () => {
    setSearchOpen((prev) => !prev);
    console.log(searchOpen);
  };
  const closeSearch = () => {
    if (searchOpen) {
      handleSearchOpening();
    }
  };

  // //Fetch from fakestoreapi
  // const APICall = () => {
  //   let tempItems = [];

  //   fetch("https://fakestoreapi.com/products")
  //     .then((res) => res.json())
  //     .then((json) => {
  //       tempItems = json;
  //     });
  //   console.log(tempItems);
  // };
  // console.log(APICall());

  useEffect(() => {
    setProducts(fakeAPIcall());
  }, []);

  useEffect(() => {}, [products]);

  const handleFilterOpening = () => {
    setIsOpen((prev) => !prev);
  };

  const handleBasketUpdate = (product) => {
    let search = trolley.find((x) => x.id === product.id);

    if (search === undefined) {
      console.log("New Item");
      setTrolley([...trolley, { ...product, quantity: 1 }]);
    } else {
      console.log("else");
      let tempTrolley = [...trolley];
      let tempSearch = tempTrolley.find((x) => x.id === product.id);
      tempSearch.quantity++;
      setTrolley([...tempTrolley]);
    }
  };

  return (
    <BrowserRouter>
      <Header
        searchOpen={searchOpen}
        trolley={trolley}
        handleSearchOpening={handleSearchOpening}
        handleFilterOpening={handleFilterOpening}
      />
      <Basket trolley={trolley} setTrolley={setTrolley} isOpen={isOpen} />
      <div onClick={closeSearch}>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route
            path="/shop"
            element={
              <Shop
                products={products}
                handleBasketUpdate={handleBasketUpdate}
              />
            }
          />
        </Routes>
      </div>
      <Footer />
    </BrowserRouter>
  );
};

export default RouteSwitch;
