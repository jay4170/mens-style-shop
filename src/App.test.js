import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import RouteSwitch from "./RouteSwitch";

describe("HomePage", () => {
  const { container } = render(<RouteSwitch />);

  it("Renders HomePage", () => {
    expect(container).toMatchSnapshot();
  });


});