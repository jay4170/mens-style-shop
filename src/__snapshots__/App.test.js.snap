// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Renders HomePage 1`] = `
<div>
  <div
    class="flex justify-between bg-cyan-50 "
  >
    <a
      href="/"
    >
      <img
        alt="DAVE on grey background"
        class="h-10"
        src="Component 1 (3).png"
      />
    </a>
    <div
      class="flex justify-between"
    >
      <a
        href="/shop"
      >
        <img
          alt="Search"
          class="h-10"
          src="search_FILL0_wght100_GRAD200_opsz48.png"
        />
      </a>
      <div>
        <img
          alt="basket"
          class="h-10"
          src="shopping_bag_FILL0_wght100_GRAD200_opsz48.png"
        />
      </div>
    </div>
  </div>
  <div
    class="bg-cyan-50 grid col-1 justify-center "
  >
    <h1
      class="font-display text-xl text-cyan-900 m-2 text-center"
    >
      MENS COMFORT AND STYLE
    </h1>
    <div
      class="h-24"
    />
    <div
      class="grid  mx-10 gap-4 justify-items-center "
    >
      <a
        href="/shop"
      >
        <div
          class="relative text-center"
        >
          <img
            alt="Full Collection"
            src="pexels-mentatdgt-1336873.jpg"
          />
          <p
            class="absolute font-display text-xl w-full  bottom-1 text-sky-500"
          >
            Full Collection
          </p>
        </div>
      </a>
      <a
        href="/shop"
      >
        <div
          class="relative text-center"
        >
          <img
            alt="Fathers Day"
            src="gb_kids_hattiebourn1x1_15.jpg"
          />
          <p
            class="absolute font-display text-xl w-full  bottom-1 text-sky-500"
          >
            Fathers Day
          </p>
        </div>
      </a>
      <a
        href="/shop"
      >
        <div
          class="relative text-center"
        >
          <img
            alt="Jewelry"
            src="Banner-No-Copy-Desktop_1920 x 480.jpg"
          />
          <p
            class="absolute font-display text-xl w-full  bottom-1 text-sky-500"
          >
            Jewelry
          </p>
        </div>
      </a>
      <a
        href="/shop"
      >
        <div
          class="relative text-center"
        >
          <img
            alt="Designer T-Shirts"
            src="pexels-monstera-5384423.jpg"
          />
          <p
            class="absolute font-display text-xl w-full  bottom-1 text-sky-500"
          >
            Designer T-Shirts
          </p>
        </div>
      </a>
    </div>
  </div>
  <footer
    class="bg-gray-200 text-center lg:text-left  bottom-0 w-full"
  >
    <div
      class="bg-sky-200 text-center p-4"
    >
      Â© 2022 Copyright: Jay
    </div>
  </footer>
</div>
`;
